tests/bundle/Functional/SearchView/Criterion/IsBookmarkedTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Tests\Bundle\Rest\Functional\SearchView\Criterion;
+
+use Ibexa\Tests\Bundle\Rest\Functional\SearchView\SearchCriterionTestCase;
+
+final class IsBookmarkedTest extends SearchCriterionTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->addMediaFolderToBookmarks();
+    }
+
+    /**
+     * @phpstan-return iterable<
+     *     string,
+     *     array{
+     *         string,
+     *         string,
+     *         int,
+     *     },
+     * >
+     */
+    public function getCriteriaPayloads(): iterable
+    {
+        yield 'Bookmarked locations' => [
+            'json',
+            $this->buildJsonCriterionQuery('"IsBookmarkedCriterion": true'),
+            1,
+        ];
+
+        yield 'Not bookmarked locations' => [
+            'json',
+            $this->buildJsonCriterionQuery('"IsBookmarkedCriterion": false'),
+            11,
+        ];
+    }
+
+    private function addMediaFolderToBookmarks(): void
+    {
+        $request = $this->createHttpRequest(
+            'POST',
+            '/api/ibexa/v2/bookmark/43'
```

Comment:
Note: relying on volatile database ID. However we'd need to rewrite this into new test-rest package tests, which is not possible for rest itself yet. Just keeping this as a note to myself here ;)